ComponentTextKit/TextKit/CKTextKitTailTruncater.mm

```diff
@@ -95,10 +95,12 @@ - (NSUInteger)_calculateCharacterIndexBeforeTruncationMessage:(NSLayoutManager *
   NSUInteger firstClippedGlyphIndex = [layoutManager glyphIndexForPoint:beginningOfTruncationMessage
                                                         inTextContainer:textContainer
                                          fractionOfDistanceThroughGlyph:NULL];
+  // If it didn't intersect with any text then it should just return the last visible character index, since the
+  // truncation rect can fully fit on the line without clipping any other text.
+  if (firstClippedGlyphIndex == NSNotFound) {
+    return [layoutManager characterIndexForGlyphAtIndex:lastVisibleGlyphIndex];
+  }
   NSUInteger firstCharacterIndexToReplace = [layoutManager characterIndexForGlyphAtIndex:firstClippedGlyphIndex];
-  CKAssert(firstCharacterIndexToReplace != NSNotFound,
```

Comment:
It's actually fine for this to be NSNotFound, it just means we don't intersect with the last line.